logging/journald_core.go

```diff
@@ -64,7 +61,7 @@ func (c *journaldCore) Write(ent zapcore.Entry, fields []zapcore.Field) error {
 	enc := zapcore.NewMapObjectEncoder()
 	c.addFields(enc, fields)
 	c.addFields(enc, c.context)
-	enc.Fields["SYSLOG_IDENTIFIER"] = c.identifier
+	enc.Fields[journaldFieldEncode("SYSLOG_IDENTIFIER")] = c.identifier
```

Comment:
`SYSLOG_IDENTIFIER` is a field specified in https://www.freedesktop.org/software/systemd/man/latest/systemd.journal-fields.html, so I don't think there's really a point in calling `journaldFieldEncode()` on it. If it would change it, it would set the wrong field afterwards. If anything, I'd put it in a constant and then add `journaldFieldEncode(fieldSyslogIdentifier) == fieldSyslogIdentifier` as a test case.

Reply:
The idea behind wrapping the field key with a `journaldFieldEncode` call was to signalize that field keys should be sanitized. I am totally aware that this static string is valid, but when someone else will alter this code in the future, they might miss the need to sanitize keys, as the other call to `journaldFieldEncode` happens in the `addFields` method.

Reply:
I have added a comment explaining the necessity to have "clean" field keys. This should be even more direct.